contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,268 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >= 0.6.0 < 0.7.0;
+
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
+
+import {SafeMath} from "./math/SafeMath.sol";
+
+contract RockPaperScissors is Ownable, Pausable {
+
+    using SafeMath for uint;
+
+    mapping(address => uint) public balances;
+    mapping(bytes32 => Game) public games;
+
+    address private constant NULL_ADDRESS = address(0);
+    string private constant ONLY_CREATOR_CALL_MSG = "Only creator can call";
+    string private constant GAME_NOT_FOUND_MSG = "Game not found";
+    string private constant INVALID_MOVE_MSG = "Invalid move";
+
+    uint256 constant FORFEIT_WINDOW = 12 hours;
+    uint256 constant FEE_PERCENTAGE = 10;
+
+    struct Game {
+        address creator;
+        uint256 stake;
+        bytes32 moveHash;
+        address opponent;
+        uint8 opponentMove;
+        uint256 expiryDate;
+    }
+
+    event Deposit(
+        address player,
+        uint256 amount
+    );
+
+    event GameCreated(
+        address indexed creator,
+        bytes32 indexed gameToken,
+        uint256 stake
+    );
+
+    event OpponentPlays(
+        address indexed opponent,
+        bytes32 indexed token,
+        uint8 move
+    );
+
+    event Draw(
+        bytes32 indexed token
+    );
+
+    event WinnerRewarded(
+        bytes32 indexed gameToken,
+        address indexed winner,
+        address indexed loser,
+        uint256 winnings,
+        uint8 creatorMove,
+        uint8 opponentMove
+    );
+
+    event WithDraw(
+        address indexed withdrawer,
+        uint amount
+    );
+
+    event  CreatorEndsGame(
+        address indexed player,
+        bytes32 indexed gameToken,
+        uint256 refund
+    );
+
+    event  ForfeitPaid(
+        address indexed player,
+        bytes32 indexed gameToken,
+        uint256 forfeit
+    );
+
+    function deposit() external payable whenNotPaused {
```

Comment:
This function is not necessary now. What it does is it advertises your contract as a potential use as a wallet. The net effect would be to potentially increase the balance it holds.

Reply:
I see.